src/jdataview.js

```diff
@@ -1,833 +1,471 @@
+					undefined,
+					undefined,
+					this.littleEndian
+			  )
+			: new jDataView(
+					this.buffer,
+					this.byteOffset + start,
+					end - start,
+					this.littleEndian
+			  );
+	}
 
-		// ArrayBuffer: we use a typed array of size 1 from original buffer if alignment is good and from slice when it's not
-		if (size === 1 || ((this.byteOffset + byteOffset) % size === 0 && littleEndian)) {
-			typedArray = new TypedArray(this.buffer, this.byteOffset + byteOffset, 1);
-			this._offset = byteOffset + size;
-			return isReadAction ? typedArray[0] : (typedArray[0] = value);
+	/**
+	 * Aligns the pointer (clearing any bitOffset). Can also move the pointer
+	 */
+	alignBy(byteCount) {
+		this.#bitOffset = 0;
+		if ((byteCount ?? 1) !== 1) {
+			return this.skip(
+				byteCount - (this.#bytePointer % byteCount || byteCount)
+			);
 		} else {
-			var bytes = new Uint8Array(isReadAction ? this.getBytes(size, byteOffset, littleEndian, true) : size);
-			typedArray = new TypedArray(bytes.buffer, 0, 1);
-
-			if (isReadAction) {
-				return typedArray[0];
-			} else {
-				typedArray[0] = value;
-				this._setBytes(byteOffset, bytes, littleEndian);
-			}
+			return this.#bytePointer;
 		}
-	},
-
-	_arrayAction: function (type, isReadAction, byteOffset, littleEndian, value) {
-		return isReadAction ? this['_get' + type](byteOffset, littleEndian) : this['_set' + type](byteOffset, value, littleEndian);
-	},
+	}
 
-	// Helpers
+	// Setters and getters
 
-	_getBytes: function (length, byteOffset, littleEndian) {
-		littleEndian = defined(littleEndian, this._littleEndian);
-		byteOffset = defined(byteOffset, this._offset);
-		length = defined(length, this.byteLength - byteOffset);
+	#getBytes(
+		length,
+		byteOffset = this.#bytePointer,
+		littleEndian = this.littleEndian
+	) {
+		length ??= this.byteLength - byteOffset;
 
-		this._checkBounds(byteOffset, length);
+		this.#checkBounds(byteOffset, length);
 
 		byteOffset += this.byteOffset;
 
-		this._offset = byteOffset - this.byteOffset + length;
-
-		var result = (
-			this._isArrayBuffer
-			? new Uint8Array(this.buffer, byteOffset, length)
-			: (this.buffer.slice || Array.prototype.slice).call(this.buffer, byteOffset, byteOffset + length)
-		);
-
-		return littleEndian || length <= 1 ? result : arrayFrom(result).reverse();
-	},
+		this.#bytePointer = byteOffset - this.byteOffset + length;
 
-	// wrapper for external calls (do not return inner buffer directly to prevent it's modifying)
-	getBytes: function (length, byteOffset, littleEndian, toArray) {
-		var result = this._getBytes(length, byteOffset, defined(littleEndian, true));
-		return toArray ? arrayFrom(result) : result;
-	},
+		const result = new Uint8Array(this.buffer, byteOffset, length);
 
-	_setBytes: function (byteOffset, bytes, littleEndian) {
-		var length = bytes.length;
-
-		// needed for Opera
-		if (length === 0) {
-			return;
-		}
+		return littleEndian || length <= 1
+			? result
+			: Array.from(result).reverse();
```

Comment:
Won't this return an inconsistent type now?

It should use `.reverse()` directly on `Uint8Array` to prevent either `Uint8Array` or `Array` being returned depending on parameter.

But now I'm also concerned that tests don't catch this - the comparisons should be verifying type of the input, and I believe deepEqual does by default, so the problem is that there is likely some extra casting before comparing values.

Reply:
It would be nice to integrate at least minimal TS checking (via `checkJS`) so that it would catch basic type issues. I know it won't work for the dynamic extension to prototype, but even if we left some `any` here and there it's still better than nothing.

Reply:
Wow, I found a bunch of tiny little errors thanks to checkJS!